x-pack/plugins/alerting/server/task_runner/task_runner.ts

```diff
@@ -708,6 +712,30 @@ export class TaskRunner<
     ruleMonitoring.execution.calculated_metrics.success_ratio =
       ruleMonitoring.execution.history.filter(({ success }) => success).length /
       ruleMonitoring.execution.history.length;
+
+    // Percentile calculations for rule executions, limited to predefined number of samples
+    const historyWithDuration = ruleMonitoring.execution.history.filter(
```

Comment:
Curious to your thoughts here.

This might grow a bit over time (that lives in the `monitoring` field) and maybe we should move this out to a separate file to make it easier to read/understand?

Reply:
I don't mind, do you mean just the percentile calculation or all monitoring related code?

Reply:
Just the calculation of the monitoring code - so this chunk as well as the chunk I added for the success ratio - or maybe we can wait until we add something more to this? more of a discussion than a recommendation